### PR TITLE
Quo2: section label

### LINKED ISSUE #17121: Quo2: Implement Section Label component
Implement this component in the quo2 component library as per the design spec.

Reuse Quo2 components wherever applicable.

Designs: https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=16927-188938&mode=design&t=zdWPPhO7J4M6J2rv-4

Acceptance Criteria:

component spec tests - https://github.com/status-im/status-mobile/blob/develop/doc/component-tests-overview.md
quo2 preview screen
export from quo2/core.cljs
following best practices - https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md

Note
Please align with design team before starting this

### PR DESCRIPTION
**Pull request by clauxx**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)
fixes #17121 

### Summary

[comment]: # (Summarise the problem and how the pull request solves it)

Adding the Section label component to Quo2
<div  style="flex-direction: row">

</div>
 

#### Platforms

- Android
- iOS

### Steps to test

- Open Status
- Open settings
- Go to `Quo2 components preview`
- Open the `settings` section
- Go to `section-label`

status: ready

### CODE REVIEW

File: src/quo2/components/settings/section_label/view.cljs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "Props:
+   - section - the label of the section
+   - description (optional) - description of the section
+   - blur? (optional) - use blurred styling"
+  [{:keys [section description blur?]}]
+  (let [theme        (quo2.theme/get-theme)
+        color        (get-text-color theme (or blur? false))
+        description? (not (nil? description))]
+    [:<>
+     [rn/text
```
Comment (J-Son89) [inline src/quo2/components/settings/section_label/view.cljs]: oh it's better to use the text component we have in quo2 components. A lot of other components are using it and we don't directly interact with the foundations.typography file other than via that component 👍

File: src/quo2/components/settings/section_label/view.cljs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                         colors/neutral-40
+                         theme)))
+
+(defn view
+  "Props:
+   - section - the label of the section
+   - description (optional) - description of the section
+   - blur? (optional) - use blurred styling"
+  [{:keys [section description blur?]}]
+  (let [theme        (quo2.theme/get-theme)
```
Comment (J-Son89) [inline src/quo2/components/settings/section_label/view.cljs]: this is not the desired way to use the theme as this interacts directly with the global ratom, rather we use a react context provider and pass it with a HOC wrapper (`with-theme`). There are many examples in the quo2 components using this approach 👍

File: src/quo2/components/settings/section_label/view.cljs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    [:<>
+     [rn/text
+      {:number-of-lines 1
+       :style           (style/section color
+                                       (if description?
+                                         :medium
+                                         :small))}
+      section]
+     (when description?
+       [rn/text {:style (style/description color)}
```
Comment (J-Son89) [inline src/quo2/components/settings/section_label/view.cljs]: same here, use quo text component, it's in `quo2.components.markdown.text`

File: src/status_im2/contexts/quo_preview/settings/section_label.cljs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    (fn []
+      [preview/preview-container
+       {:state                     state
+        :descriptor                descriptor
+        :component-container-style {:align-items :center}}
+       [rn/view
+        {:style {:flex          1
+                 :align-self    :stretch
+                 :border-radius 24}}
+        [preview/blur-view
```
Comment (J-Son89) [inline src/status_im2/contexts/quo_preview/settings/section_label.cljs]: @OmarBasem is this the  preferred way to use `blur` in these preview screens? 🤔

File: src/quo2/components/settings/section_label/view.cljs (hunk 2 above)
Comment (J-Son89) [reply src/quo2/components/settings/section_label/view.cljs]: https://github.com/status-im/status-mobile/blob/develop/src/quo2/README.md#themes

File: src/status_im2/contexts/quo_preview/settings/section_label.cljs (hunk 4 above)
Comment (OmarBasem) [reply src/status_im2/contexts/quo_preview/settings/section_label.cljs]: Nope, we use it like that: 
```
[preview/preview-container
       {:state                 state
        :descriptor            descriptor
        :blur?                 (:blur? @state)
        :show-blur-background? true}
```

Comment (clauxx) [reply src/status_im2/contexts/quo_preview/settings/section_label.cljs]: oops, didn't notice 👁️ . Thanks!

File: src/quo2/components/settings/section_label/view.cljs (hunk 2 above)
Comment (clauxx) [reply src/quo2/components/settings/section_label/view.cljs]: Thanks! Didn't see this README before 😊

**Comment by clauxx** [comment]
@J-Son89 can you give this another look please?

File: src/quo2/core.cljs [hunk 5]
```diff
@@ -312,6 +313,7 @@
 (def reorder-item quo2.components.settings.reorder-item.view/reorder-item)
 (def category quo2.components.settings.category.view/category)
 (def data-item quo2.components.settings.data-item.view/view)
+(def section-label quo2.components.settings.section-label.view/view)
```
Comment (J-Son89) [inline src/quo2/core.cljs:319]: But could you sort these settings exports here? :)

File: src/status_im2/contexts/quo_preview/settings/section_label.cljs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    (fn []
+      [preview/preview-container
+       {:state                 state
+        :descriptor            descriptor
+        :show-blur-background? @blur?
+        :blur?                 @blur?
+        :blur-height           150}
+       [quo/section-label
+        {:section     @section
+         :description (if (empty? @description)
```
Comment (J-Son89) [inline src/status_im2/contexts/quo_preview/settings/section_label.cljs]: But: (when-not (empty? @description) @description))

**Review by J-Son89 — APPROVED** [review]

**Comment by clauxx** [comment]
@J-Son89 done

**Comment by clauxx** [comment]
@Francesca-G Ready for the design review. From the docs I saw that new quo2 components don't need QA tests, so moved it manually to the Design Review column.

**Comment by Francesca-G** [comment]
> @Francesca-G Ready for the design review. From the docs I saw that new quo2 components don't need QA tests, so moved it manually to the Design Review column.

I can't seem to find the current build to test the iOS PR, can you please link it? 
Let me know if it's a problem on my side too :)

**Comment by clauxx** [comment]
> > @Francesca-G Ready for the design review. From the docs I saw that new quo2 components don't need QA tests, so moved it manually to the Design Review column.
> 
> I can't seem to find the current build to test the iOS PR, can you please link it? Let me know if it's a problem on my side too :)

I'm not sure :) Where do you usually get it from?

**Comment by churik** [comment]
@clauxx 

if the component is not used in the current app - yes, it may skip manual QA process and in this case you don't need to use `request-manual-qa` label
However, e2e tests should be launched and reviewed for all PRs.

So, please do not pass `E2E tests` column, thanks!

**Comment by clauxx** [comment]
> @clauxx
> 
> if the component is not used in the current app - yes, it may skip manual QA process and in this case you don't need to use `request-manual-qa` label However, e2e tests should be launched and reviewed for all PRs.
> 
> So, please do not pass `E2E tests` column, thanks!

Thanks! Will know next time 👍

**Comment by clauxx** [comment]
> > @Francesca-G Ready for the design review. From the docs I saw that new quo2 components don't need QA tests, so moved it manually to the Design Review column.
> 
> I can't seem to find the current build to test the iOS PR, can you please link it? Let me know if it's a problem on my side too :)

It should be available now. I guess as soon as it finishes E2E tests.

**Comment by churik** [comment]
all E2E failures are known and not related to current PR

**Review by Francesca-G — APPROVED** [review]
Here's the [review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3506%3A249148&mode=design&t=hsLLSGxFsH38BCS2-1) with a minor fix ✨

Also, I was able to select the light blur option for this component even though it's not supposed to have it (only dark blur should be allowed)

**Comment by clauxx** [comment]
@Francesca-G fixed it here already. Please have a look.

**Review by Francesca-G — APPROVED** [review]

**Pull request merged**